controllers/noderefutil/providerid.go

```diff
@@ -15,6 +15,8 @@ limitations under the License.
 */
 
 // Package noderefutil implements NodeRef utils.
+// The ProviderID type is deprecated and unused by Cluster API internally.
+// It will be removed entirely in a future release.
 package noderefutil
```

Comment:
I think we should also deprecate the exported types so that linters pick it up
Should work roughly like this:
```go
// Deprecated: This field has no function and is going to be removed in a next release.
```

Reply:
Good call, added `// Deprecated`... comments for all exported matter in this file.